Клиентская программа

stepan-dolgorukov opened the pull request.

stepan-dolgorukov committed:
Клиент-база

stepan-dolgorukov committed:
Примитивное получение аргументов

stepan-dolgorukov committed:
Указание IP-адреса сервера

stepan-dolgorukov committed:
Исходный текст клиента помещён в каталог

stepan-dolgorukov committed:
Прочитайка для клиента

stepan-dolgorukov committed:
Обработка аргументов программы

stepan-dolgorukov committed:
Функция, показывающая как пользоваться программой

stepan-dolgorukov committed:
Обновление прочитайки

LifeStyle28 commented on Client/client.cxx:
```diff
@@ -0,0 +1,54 @@
+  prop::notify(varmap);
+
+  if (varmap.count("help")) {
+    show_how_to_use();
```
desc тоже выводится в поток, и можно не писать отдельную функцию
1 аргумент у add_options это сама команда, причем можно через запятую писать сокращенное применение, допустим "help,h"
2 аргумент значение, допустим если мы считаем какие-то тики, то там будут миллисекунды
3 аргумент детальное описание

LifeStyle28 commented on Client/client.cxx:
```diff
@@ -0,0 +1,54 @@
+  asio::io_context context{};
+
+  const auto server_addr{asio::ip::make_address("192.168.0.10")};
+  asio::ip::udp::endpoint endpoint{server_addr, 3333u};
```
здесь я хотел, чтобы не только команда была как параметр командной строки, но еще и ИП-адрес сервера и порт. Но в целом, я думаю, что ты понимаешь как это сделать

stepan-dolgorukov replied:
Оформил считывание адреса и порта

LifeStyle28 reviewed (approved):
Эту работу тоже зачту, попробуй поиграть с boost::program_options

stepan-dolgorukov committed:
Распознавание случая неправильного синтаксиса передачи аргументов

Например, закрыл случай, когда передаётся --command, а значение не
указывается.

stepan-dolgorukov committed:
Разбор аргументов: IP-адрес & Порт

stepan-dolgorukov committed:
Короткие ключи

stepan-dolgorukov committed:
Функция показа доступных команд серверу

stepan-dolgorukov committed:
Вывод desc-а на экран вместе с доступными командами

stepan-dolgorukov committed:
Ненулевой код возврата при отсутствии нужных аргументов
